File: csharp/isbn-verifier/IsbnVerifier.cs
using System;
using System.Text.RegularExpressions;
using System.Linq;

public static class IsbnVerifier
{
    public static bool IsValid(string number)
    {
        Regex isbnPattern = new Regex(@"^\d-?\d{3}-?\d{5}-?[\dX]$");

        if (!isbnPattern.IsMatch(number))
            return false;

        int sum = number.Replace("-", "")
                     .Replace("X", "A")
                     .Select((c, idx) => Convert.ToInt32(c.ToString(), 16) * (10 - idx))
                     .Sum();

        return sum % 11 == 0;


    }
}

File: python/isbn-verifier/isbn_verifier.py
import re

def is_valid(isbn: str) -> bool:
    isbn_pattern = r'^\d-?\d{3}-?\d{5}-?[\dX]$'

    if not re.match(isbn_pattern, isbn):
        return False

    clear_ISBN_in_hex = isbn.replace('-', '').replace('X', 'A')
    summary = sum([int(value, base=16) * ( 10 - index) for index, value in enumerate(clear_ISBN_in_hex)]) 
    return summary % 11 == 0
